ap.md

```diff
@@ -0,0 +1,236 @@
+
+<hr>
+
+Let's start this next part by building out the web page we want Tessel to send to your web browser. Create a file called `index.html` in your project directory and open it up in your preferred text editor to add the initial html for our web page:
+
+{% highlight html %}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Tessel Web App</title>
+</head>
+<body>
+  <h1>Hello from Tessel!</h1>
+  <p>Press a button to toggle an LED.</p>
+  <ul>
+    <li>
+      <button class="led-button" data-led="2">Green</button>
+      Status: <span class="led-status"></span>
+    </li>
+    <li>
+      <button class="led-button" data-led="3">Blue</button>
+      Status: <span class="led-status"></span>
+    </li>
+  </ul>
+</body>
+</html>
+{% endhighlight %}
+
+If you open that file in your browser, you should see something like this:
+
+![web app screenshot](./images/ap-web-app-preview.png)
+
+In that html, we have a heading, a line of instructions for the user, and a list of buttons for controlling a corresponding LED on the Tessel. Below that list of buttons we'll add some JavaScript for communicating with the Tessel:
+
+{% highlight html %}
+<script type="text/javascript">
+  // Get a NodeList of elements with the class 'led-button'
+  var buttons = document.querySelectorAll('.led-button');
+
+  // Iterate through that Nodelist and add a 'click' EventListener
+  Array.prototype.forEach.call(buttons, function (button) {
+    button.addEventListener('click', toggleLed);
+  });
+
+  // Our event handler function for 'click' events on the LED buttons
+  function toggleLed (event) {
+    var button = event.target;
+    var ledIndex = button.getAttribute('data-led'); // The index of the led in the Tessel.led array
+    var statusNode = button.parentNode.querySelector('.led-status'); // The sibling status span to update
+
+    // create a new XHR for communicating requests to our Tessel server
+    var req = new XMLHttpRequest();
+
+    // Open a GET request to '/leds/:index'
+    req.open('GET', '/leds/' + ledIndex);
+
+    // Once the request gets a successful response, update that statusNode with the status of the LED.
+    req.onload = function(e) {
+      if (req.readyState == 4 && req.status == 200) {
+        if (req.status == 200) {
+          var response = JSON.parse(req.responseText);
+          statusNode.textContent = response.on ? 'ON' : 'OFF';
+        } else { 
+          console.log('Error'); 
+        }
+      }
+    }
+    req.send(); // send our request to the server
+  }
+</script>
+{% endhighlight %}
+
+
+While there are some comments in the previous code snippet, the summary is when a button with the class "led-button" is clicked, we grab the info from its "data-led" attribute and make a request to our Tessel server with that info in the url. After we get a successful response from the server, we update the "Status" field next to the button with the state of the corresponding LED. 
```

Comment:
Convention in this tutorial is to just use the comments and not also summarize - or if it really needs a summary, it should be in comments at the head of the code.

I personally prefer the comment style because it makes people read the code rather than skim over it to the summary.